# Perdidos en el espacio

Imagina que en el futuro una nave espacial tripulada parte de la Tierra hacia el planeta Neptuno.

Durante el viaje, la tripulación permanece en estado de animación suspendida. En el trayecto la nave
sufre una grave avería y el sistema automático de control la dirige al cuerpo celeste más cercano,
donde realiza un aterrizaje de emergencia. Al posarse sobre la superficie, el sistema de soporte vital
de la nave reanima a la tripulación.

Una vez despiertos, los tripulantes descubren que el sistema de posicionamiento espacial no
funciona y por el momento no tienen comunicación con la Tierra ni pueden mirar al exterior de la
nave por razones de seguridad. Así que no saben dónde están, no tienen ningún dato del exterior,
ni posibilidad de conseguirlo, ni saben cuánto tiempo ha pasado.

En su trayecto entre la Tierra y Neptuno han podido estar cerca de la propia Tierra, Marte, Júpiter,
Saturno, Urano, Neptuno y las lunas de cualquiera de ellos

In [1]:
import random
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from faker import Faker
from sklearn.preprocessing import LabelEncoder

# Datos Experimentales


In [7]:
filepath = 'dataset.csv'
dataset = pd.read_csv(filepath)

dataset.head(100)

ID  Longitud (m)  Tiempo (s)  g_calculado (m/s²) Planeta más probable  \
0   1          1.00        0.46                9.48         Tierra (95%)   
1   2          2.00        0.66                9.18        Saturno (60%)   
2   3          1.10        0.48                9.50         Tierra (96%)   
3   4          1.25        0.50               10.00         Tierra (90%)   
4   5          1.50        0.55                9.92         Tierra (98%)   
5   6          1.75        0.59               10.08         Tierra (92%)   
6   7          1.60        0.56               10.20         Tierra (85%)   
7   8          2.20        0.67                9.75         Tierra (75%)   
8   9          1.30        0.57                8.02        Saturno (90%)   
9  10          1.05        0.47                9.50         Tierra (96%)   

  2do más probable 3ro más probable  
0     Saturno (5%)       Urano (0%)  
1     Tierra (40%)       Urano (0%)  
2     Saturno (4%)       Urano (0%)  
3    Saturno (10%)       Urano (0%)  
4     Saturno (2%)       Urano (0%)  
5     Saturno (8%)       Urano (0%)  
6    Saturno (15%)       Urano (0%)  
7    Saturno (25%)       Urano (0%)  
8      Urano (10%)      Tierra (0%)  
9     Saturno (4%)       Urano (0%)

# Fórmula: L = 1/2 x g x t^2

    A.- Con una longitud L= 1.00 m cronometran t = 0.46 s.

    ¿Cuánto vale g y su error?
 
    ¿En qué planeta están, o pueden estar?

In [8]:
import numpy as np

L = 1.00  # m
t = 0.46  # s
delta_L = 0.01  # m
delta_t = 0.01  # s

g = 2 * L / t**2

dg_dL = 2 / t**2
dg_dt = -4 * L / t**3

delta_g = np.sqrt((dg_dL * delta_L)**2 + (dg_dt * delta_t)**2)

g, delta_g

df = pd.DataFrame({'Aceleración de la gravedad medida (g)': [g],
                   'Error en la medición de g': [delta_g]})

print(df)

   Aceleración de la gravedad medida (g)  Error en la medición de g
0                               9.451796                   0.421677


Dado que g = 9.45 m/s^2 + - 0.42 m/s², podemos calcular el rango de valores posibles para g:

Rango inferior: 9.45 - 0.42 = 9.03 m/s^2

Rango superior: 9.45 + 0.42 = 9.87 m/s^2

Teniendo en cuenta los datos de la tabla, podemos saber que la tripulación podría estar en:

Saturno (g = 9.0 m/s^2) ya que 9.0 está dentro del rango de 9.03 a 9.87 m/s^2.

Tierra (g = 9.8 m/s^2) ya que 9.8 también está dentro del rango de 9.03 a 9.87 m/s^2.

# Creación de un Dataset enfocado en Experimentos de Aceleración de Gravedad para determinar la ubicación de una nave espacial en Tierra o Saturno 

Ahora vamos a crear un dataset enfocado en los experimentos realizados en los planetas con mayores porcentajes de posibilidad de estar presentes. Estos experimentos se centran en la Tierra y Saturno, ya que, según los cálculos previos, estos dos planetas mostraron los valores de aceleración de la gravedad más cercanos a los obtenidos en los escenarios analizados.

Este dataset contendrá los datos de las longitudes y tiempos cronometrados de cada experimento, así como los valores de la aceleración de la gravedad calculados y los porcentajes de acierto para cada planeta. Al analizar estos datos, podremos obtener una mejor comprensión de las posibilidades de que la nave espacial se encuentre en la Tierra o en Saturno, y así tomar decisiones más informadas sobre cómo proceder en esta situación.

Al utilizar este enfoque centrado en los planetas con mayores probabilidades, esperamos que podamos obtener resultados más precisos y útiles para determinar la ubicación de la nave espacial y garantizar el éxito de la misión.

In [9]:
filepath = 'dataset(tierra,saturno).csv'
data = pd.read_csv(filepath)
data.head(100)


ID  Longitud  Tiempo  g_calculado Planeta_mas_probable  2do_mas_probable
0   1       0.8    0.40        10.00               Tierra                80
1   2       1.3    0.52         9.52               Tierra                93
2   3       1.5    0.56         9.57               Tierra                91
3   4       0.9    0.43         9.67               Tierra                88
4   5       1.1    0.47         9.97               Tierra                82
5   6       2.1    0.66         9.60               Tierra                89
6   7       1.8    0.61         9.63               Tierra                87
7   8       1.2    0.50         9.60               Tierra                89
8   9       2.3    0.69         9.53               Tierra                92
9  10       1.4    0.54         9.55               Tierra                90